torch/_inductor/codegen/cpp.py

```diff
@@ -1706,10 +1796,17 @@ def gen_loop(loop: LoopLevel, in_reduction=False):
                     stack.enter_context(code.indent())
                     # generate inner loops or loop body
                     if loop.inner:
-                        gen_loops(loop.inner, loop.is_reduction())
+                        if update_reduction_store and loop.parallel:
+                            for inner in loop.inner:
+                                inner.get_kernels()[
+                                    0
+                                ].update_stores_with_parallel_reduction()
```

Comment:
Can you leave the update call until before `gen_kernel` below?

Reply:
Yes, done